hacks/prune-rhcos-snapshots/prune-rhcos-snapshots.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+import os
+import re
+
+import boto3
+import openshift as oc
+import re
+import json
+import yaml
+from typing import Dict, List, Set
+import datetime
+import urllib3
+import pathlib
+import subprocess
+
+# Set to "true" if an AMI is selected for garbage collection
+AMI_TAG_KEY_GARBAGE_COLLECT = 'garbage_collect'
+
+# Set to "true" if an AMI is mentioned in installer reference
+AMI_TAG_KEY_PRODUCTION = 'production'
+
+
+def create_recycle_bin_rule(client, res_type: str):
+    response = client.create_rule(
+        RetentionPeriod={
+            'RetentionPeriodValue': 60,
+            'RetentionPeriodUnit': 'DAYS'
+        },
+        Tags=[
+            {
+                'Key': 'Name',
+                'Value': f'rhcos-preserve-{res_type}'
+            },
+        ],
+        Description=f'Help protect RHCOS {res_type} from accidental deletion',
+        ResourceType=res_type,
+    )
+    print(f'Created new retention rule: {response["Identifier"]}')
+
+
+Commit = str
+Region = str
+AmiId = str
+
+if __name__ == '__main__':
+
+    gopath = pathlib.Path(os.getenv('GOPATH'))
+    if not gopath.is_dir():
+        print('Set GOPATH env before running')
+        exit(1)
+
+    installer_git_path = gopath.joinpath('src/github.com/openshift/installer')
+    if not installer_git_path.is_dir():
+        print(f'Requires recent clone of github.com/openshift/installer in GOPATH: {str(installer_git_path)}')
+        exit(1)
+
+    http = urllib3.PoolManager()
+
+    amis_in_use: Set = set()
+
+    git_depth = 0
+    while True:
+        installer_commit: Commit = f'upstream/master~{git_depth}'
```

Comment:
This is going down `master` only. I think we need to go down from all the release branches too, no?

Reply:
Great call. I wasn't considering post-cut changes. Updated.